test/action.yaml

```
@@ -25,26 +25,36 @@ runs:
     - name: Checkout
       uses: actions/checkout@v3
       if: inputs.checkout-repo
+    - name: Verify needed scripts exist
+      id: verify-scripts-exist
+      shell: bash
+      run: |
+        if [[ -f ./script/setup && -f ./script/test ]]; then
+          echo "::set-output name=needed-scripts-exist::true"
+          echo "Needed scripts do exist"
+        else
+          echo "::set-output name=needed-scripts-exist::false"
+          echo "Needed scripts do NOT exist"
```

All the future steps depend on this being `true` - so why not fail here?

```suggestion
          echo "::error::Required script/setup or script/test is missing or not executable"
          exit 1
```

---

I read [here](https://stackoverflow.com/questions/60589373/how-to-force-to-exit-in-github-actions-step) that there is no way to arbitrarily exit a job. That exit as part of a run would just exit the step, not the job correct? That documentation is not GHA official documentation though.

---

By default a failed step does not run any further steps in the job unless they have `if: failure()` as a clause.